File: solutions/julia/armstrong-numbers/1/armstrong-numbers.jl
function isarmstrong(num)
    if (num<=0) 
        num==0 
    else
        digits = []
        n = num
        while n>0
            d = n % 10 
            push!(digits, d)
            n = (n - d) /10
        end
        sum(x->x^length(digits),digits) == num  
    end
end 

File: solutions/fortran/armstrong-numbers/1/armstrong_numbers.f90

module armstrong_numbers
  implicit none
contains

  logical function isArmstrongNumber(i)
    integer, intent(in) :: i
    integer:: j, k, mysum, ten_power
    integer, dimension(1:9) :: digits

    if (i==0) then
      isArmstrongNumber = .true.
    else
      digits = (/ 0,0,0,0,0,0,0,0,0 /)
      j = i
      k = 1
      do while (j>0)
        digits(k) = mod(j,10)
        j = (j - mod(j,10))/10
        k = k+1
      end do
      mysum = sum(digits**(k-1))
      isArmstrongNumber = (mysum == i)
    end if
  end function

end module
